### PR TITLE
Fix/reduce llm calls

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

* Only calls LLM when necessary
* Fixes a bug where the proposals where not updated during NO_VOTE events

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

**Review by 0xArdi — APPROVED** [review]

File: packages/valory/skills/proposal_collector_abci/rounds.py [hunk 1]
```diff
@@ -74,7 +74,12 @@ def proposals(self) -> dict:
     @property
     def votable_proposal_ids(self) -> set:
         """Get the votable proposal ids, sorted by their remaining blocks until expiration, in ascending order."""
-        return set(self.db.get("votable_proposal_ids", {}))
+        return cast(set, self.db.get("votable_proposal_ids", {}))
```
Comment (Adamantios) [inline packages/valory/skills/proposal_collector_abci/rounds.py:77]: Why that change?

File: packages/valory/skills/proposal_collector_abci/rounds.py [hunk 2]
```diff
@@ -135,10 +142,18 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
                 if not existing:
                     delegations.append(nd)
 
+                # Do we need to refresh any proposal?
+                for p in proposals.values():
+                    if nd["token_address"] in p["governor"]["tokens"][0]["id"]:
+                        proposals_to_refresh.add(p["id"])
```
Comment (Adamantios) [inline packages/valory/skills/proposal_collector_abci/rounds.py:148]: It looks like this logic should belong to some behaviour.

File: packages/valory/skills/proposal_collector_abci/rounds.py (hunk 1 above)
Comment (dvilelaf) [reply packages/valory/skills/proposal_collector_abci/rounds.py:77]: Linting issues. Would you like me to revert it?

File: packages/valory/skills/proposal_collector_abci/rounds.py (hunk 2 above)
Comment (dvilelaf) [reply packages/valory/skills/proposal_collector_abci/rounds.py:148]: I think we should leave it here: we need access to the different new delegations from every agent (this round is CollectDifferent), which  we aggregate here.

Comment (Adamantios) [reply packages/valory/skills/proposal_collector_abci/rounds.py:148]: I see, it makes sense to keep this here as we avoid an extra round.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**